test/powershell/engine/Basic/Telemetry.Tests.ps1

```diff
@@ -0,0 +1,137 @@
+        $env:POWERSHELL_TELEMETRY_OPTOUT = "no"
+        & $PWSH -command "exit"
+        Test-Path -Path $uuidPath  | Should -Be $true
+    }
+
+    It "Should create a uuid file by default" {
+        if ( Test-Path env:POWERSHELL_TELEMETRY_OPTOUT ) { Remove-Item -Path env:POWERSHELL_TELEMETRY_OPTOUT }
+        & $PWSH -command "exit"
+        Test-Path -Path $uuidPath  | Should -Be $true
+    }
+
+    It "Should create a property uuid file when telemetry is sent" {
+        $env:POWERSHELL_TELEMETRY_OPTOUT = "no"
+        & $PWSH -command "exit"
+        Test-Path -Path $uuidPath  | Should -Be $true
+        (Get-ChildItem -Path $uuidPath).Length | Should -Be 16
+        [byte[]]$newBytes = Get-Content -AsByteStream -Path $uuidPath
+        [System.Guid]::New($newBytes) | Should -BeOfType [System.Guid]
+    }
+
+    It "Should not create a telemetry file if one already exists and telemetry is opted in" {
+        [byte[]]$bytes = [System.Guid]::NewGuid().ToByteArray()
+        [System.IO.File]::WriteAllBytes($uuidPath, $bytes)
+        & $PWSH -command "exit"
+        [byte[]]$newBytes = Get-Content -AsByteStream -Path $uuidPath
+        Compare-Object -ReferenceObject $bytes -DifferenceObject $newBytes | Should -BeNullOrEmpty
+    }
+
+    It "Should create a new telemetry file if the current one is 00000000-0000-0000-0000-000000000000" {
+        [byte[]]$zeroGuid = [System.Guid]::Empty.ToByteArray()
+        [System.IO.File]::WriteAllBytes($uuidPath, $zeroGuid)
+        & $PWSH -command "exit"
+        [byte[]]$newBytes = Get-Content -AsByteStream -Path $uuidPath
+        # we could legitimately have zeros in the new guid, so we can't check for that
+        # we're just making sure that there *is* a difference
+        Compare-Object -ReferenceObject $zeroGuid -DifferenceObject $newBytes | Should -Not -BeNullOrEmpty
+    }
+
+    It "Should create a new telemetry file if the current one is smaller than 16 bytes" {
+        $badBytes = [byte[]]::new(8);
+        [System.IO.File]::WriteAllBytes($uuidPath, $badBytes)
+        & $PWSH -command "exit"
+        [byte[]]$nb = Get-Content -AsByteStream -Path $uuidPath
+        [System.Guid]::New($nb) | Should -BeOfType [System.Guid]
+    }
+
+    It "Should not create a new telemetry file if the current one has a valid guid and is larger than 16 bytes" {
+        $g = [Guid]::newGuid()
+        $tooManyBytes = $g.ToByteArray() * 2
+        [System.IO.File]::WriteAllBytes($uuidPath, $tooManyBytes)
+        [byte[]]$nb = Get-Content -Path $uuidPath -AsByteStream | Select-Object -First 16
+        $ng = [System.Guid]::new($nb)
+        $g | Should -Be $ng
+    }
+
+
+    It "Should properly set whether telemetry is sent based on when environment variable is not set" -TestCases $telemetryIsSetData {
+        param ( [string]$name, [string]$value, [string]$expectedValue )
+        if ( Test-Path -Path env:POWERSHELL_TELEMETRY_OPTOUT ) {
+            Remote-Item -Path env:POWERSHELL_TELEMETRY_OPTOUT
```

Comment:
Is this needed, `AfterEach` seems to be doing this.

Reply:
actually, this test had a more serious problem as telemetryIsSetData wasn't actually assigned when we got to this test. I've rewritten it